waiter/src/waiter/util/client_tools.clj

```diff
@@ -148,6 +148,14 @@
   (not= (retrieve-waiter-port waiter-url)
         (retrieve-h2c-port waiter-url)))
 
+(defn grpc-cancellations-supported?
+  "Returns true if gRPC cancellations are supported.
+   Certain proxies, e.g. haproxy https://github.com/haproxy/haproxy/issues/172, do not support cancellations.
+   Other proxies, e.g. Envoy, do support gRPC cancellations."
+  [waiter-url]
+  (or (not (behind-proxy? waiter-url))
+      (not (str/blank? (System/getenv "WAITER_GRPC_CLEARTEXT_PORT")))))
```

Comment:
Can you make the connection between the implementation and the docs for this fn?

Reply:
Improved the doctring